tools/run_examples.py

```diff
@@ -0,0 +1,91 @@
+"""
+Run and time some or all examples.
+"""
+
+from argparse import ArgumentParser
+from contextlib import ExitStack
+import os
+from pathlib import Path
+import subprocess
+import sys
+from tempfile import TemporaryDirectory
+import time
+
+
+_preamble = """\
+from matplotlib import pyplot as plt
+
+def pseudo_show(block=True):
+    for num in plt.get_fignums():
+        plt.figure(num).savefig(f"{num}")
+
+plt.show = pseudo_show
+
+"""
+
+
+class RunInfo:
+    def __init__(self, backend, elapsed, failed):
+        self.backend = backend
+        self.elapsed = elapsed
+        self.failed = failed
+
+    def __str__(self):
+        s = ""
+        if self.backend:
+            s += f"{self.backend}: "
+        s += f"{self.elapsed}ms"
+        if self.failed:
+            s += "(failed!)"
+        return s
+
+
+def main():
+    parser = ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "--backend", action="append",
+        help=("backend to test; can be passed multiple times; defaults to the "
+              "default backend"))
+    parser.add_argument(
+        "--include-sgskip", action="store_true",
+        help="do not filter out *_sgskip.py examples")
+    parser.add_argument(
+        "--rundir", type=Path,
+        help=("directory from where the tests are run; defaults to a "
+              "temporary directory"))
+    parser.add_argument(
+        "paths", nargs="*", type=Path,
+        help="examples to run; defaults to all examples (except *_sgskip.py)")
+    args = parser.parse_args()
+
+    root = (Path(__file__).parent / "../examples").resolve()
```

Comment:
`.parent` and `..`?

Reply:
the first parent is basically dirname(); I guess the other option would be `.parents[1] / "examples"` but I always find it hard to get the right index in the brackets...

Reply:
Could do `.parent.parent`, maybe?

Reply:
sure, that's fine too